Feature/39 db layer for product service

stishchenko opened the pull request.

stishchenko committed:
#39 Creation of mapper
ProductMapper -> create interface.
ProductMapper.xml -> create base implementation of interface ProductMapper. [FEATURE] need to refactor.

stishchenko committed:
#39 Refactoring of mapper
AppExceptionHandler -> add printing exceptions to console.
ProductDaoImpl -> add usage of ProductMapper.
ProductMapper -> change arg of delete on id.
ProductMapper.xml -> describe result map and queries. [BUG] need to refactor.

stishchenko committed:
#39 Finish refactor ProductMapper
ProductDaoImpl -> remove getProduct().
ProductMapper -> change return type.
ProductMapper.xml -> resolve errors.

stishchenko committed:
Merge branch 'master' into feature/39_DB_layer_for_ProductService

# Conflicts:
#	src/main/java/team/asd/dao/ProductDaoImpl.java
#	src/main/java/team/asd/exception/AppExceptionHandler.java

rdovgan commented on src/main/java/team/asd/dao/ProductDaoImpl.java:
```diff
@@ -1,39 +1,44 @@
 	public Product saveProduct(Product product) {
-		return product;
+		productMapper.insertProduct(product);
+		return productMapper.readProductById(product.getId());
```
@stishchenko no need to read product. You can set `void` for this method

rdovgan commented on src/main/java/team/asd/dao/ProductDaoImpl.java:
```diff
@@ -1,39 +1,44 @@
 	public Product updateProduct(Product product) {
-		return product;
+		productMapper.updateProduct(product);
+		return productMapper.readProductById(product.getId());
```
@stishchenko The same here

rdovgan commented on src/main/java/team/asd/dao/ProductDaoImpl.java:
```diff
@@ -1,39 +1,44 @@
 	@Override
+	@Transactional
 	public Integer deleteProduct(Integer id) {
+		productMapper.deleteProduct(id);
```
For `delete` method we should set `void` or `boolean`

rdovgan reviewed (changes requested):
@stishchenko please check my comments

stishchenko committed:
#39
Change return types of create, update, delete methods on void, replace returning.
